frame/vamm/src/tests/swap_base_asset.rs

```diff
@@ -0,0 +1,115 @@
+use crate::{
+	mock::{ExtBuilder, MockRuntime, TestPallet},
+	pallet::Error,
+	tests::{
+		balance_range_lower_half, balance_range_upper_half, get_swap_config, get_vamm_state,
+		TestSwapConfig, RUN_CASES,
+	},
+};
+use composable_traits::vamm::{AssetType, Direction, Vamm as VammTrait};
+use frame_support::{assert_noop, assert_ok};
+use proptest::prelude::*;
+
+proptest! {
+	#![proptest_config(ProptestConfig::with_cases(RUN_CASES))]
+	#[test]
+	fn swap_base_remove_insufficient_funds_error(
+		mut vamm_state in get_vamm_state(Default::default()),
+		mut swap_config in get_swap_config(
+			TestSwapConfig {
+				direction: Some(Direction::Remove),
+				vamm_id: Some(0),
+				asset: Some(AssetType::Base),
+				..Default::default()}),
+		input_amount in balance_range_upper_half(),
+		base_asset_reserves in balance_range_lower_half(),
+	) {
+		prop_assume!(input_amount > base_asset_reserves);
```

Comment:
I think you can get rid of this by generating `input_amount` and `base_asset_reserves` jointly in a `prop_compose`. Should make shrinking inputs better in case errors are found

Reply:
actually, in the way `range_upper` and `range_lower` were made it would be technically impossible to create values that invalidate this `prop_assume`, so this assert-like statement there is just to be pedantic about the requirement we're expecting for this proptest.

Do you think I should refactor it tho?

Reply:
Ah, I did not check the definition of those functions. Well, if by definition they satisfy the property in line 27, then I think removing the `prop_assume!` is warranted.

I would still prefer to generate then jointly, though, to avoid errors in the future if you change the implementation of either of two functions. Your call really